dfdaemon/config/config.go

```diff
@@ -347,7 +348,7 @@ func certPoolFromFiles(files ...string) (*x509.CertPool, error) {
 
 	roots := x509.NewCertPool()
 	for _, f := range files {
-		cert, err := ioutil.ReadFile(f)
+		cert, err := afero.ReadFile(fs, f)
```

Comment:
Could you explain why do you use the function in 'spf13/afero' package rather than golang internal package? Is there any advantage?

Reply:
[spf13/afero](https://github.com/spf13/afero) abstract the file system, so we can mock the file system with `afero.NewMemFs` during unit tests.

 It removes the host file system as test dependency, and it's easier to setup and teardown. Mocking cases where something does not exists is much easier.

Reply:
@fengzixu  Using the standard library is fine and unit tests can be done with temp dir. I'm ok to revert this and write tests using the the actual file system, since other parts of the code is using the fs directly. Maybe we can ask the opinion from other members? @lowzj @Starnop

Reply:
I prefer to use one testing framework in Dragonfly. Maybe `testify` is better than `go-check` used now. But recently we're going to release `0.4.0` and there's no plan to change a testing framework in this version. 
We can change the testing framework in next version after discussion. 
WDYT @inoc603 @fengzixu @Starnop

Reply:
@lowzj the `client` package uses `testify` too, and I thought we use both testing framework in Dragonfly, so I chose one from them. If we want to stick to `go-check`, we should change tests in `client` package too. The 2 frameworks are very similar in many ways, I think it's fine they co-exist in one project.

Personally I would prefer `testify`. It's more popular, the development is more active, and it's more flexible that it does not force you to organize your tests in a particular way. I use `suite.Suite` because it's similar to the way `go-check` organize tests, but you can use `testify` in standard go tests as an assertion library only.